appeals/web/src/server/appeals/appeal-details/interested-party-comments/add-ip-comment/add-ip-comment.controller.js

```diff
@@ -217,7 +220,65 @@ export async function postDateSubmitted(request, response) {
  * @param {import('@pins/express/types/express.js').RenderedResponse<any, any, Number>} response
  */
 export async function postIPComment(request, response) {
-	response.end();
+	const documentGuid = request.session.fileUploadInfo?.files[0].GUID
+	const currentFolder = request.session.fileUploadInfo?.folderId
+
+	try {
+		const {
+			currentAppeal,
+			session: { fileUploadInfo }
+		} = request;
+
+		/** @type {import('@pins/appeals/index.js').AddDocumentsRequest} */
+		const addDocumentsRequestPayload = {
+			blobStorageHost:
+				config.useBlobEmulator === true ? config.blobEmulatorSasUrl : config.blobStorageUrl,
+			blobStorageContainer: config.blobStorageDefaultContainer,
+			documents: fileUploadInfo.files.map(
+				(/** @type {import('#lib/ts-utilities.js').FileUploadInfoItem} */ document) => {
+					/** @type {import('@pins/appeals/index.js').MappedDocument} */
+					const mappedDocument = {
+						caseId: currentAppeal.appealId,
+						documentName: document.name,
+						documentType: document.documentType,
+						mimeType: document.mimeType,
+						documentSize: document.size,
+						stage: document.stage,
+						folderId: currentFolder,
+						GUID: document.GUID,
+						receivedDate: document.receivedDate,
+						redactionStatusId: document.redactionStatus,
+						blobStoragePath: document.blobStoreUrl
+					};
+
+					return mappedDocument;
+				}
+			)
+		};
+
+		await createNewDocument(request.apiClient, currentAppeal.appealId, addDocumentsRequestPayload);
+
+		await createIPComment(
+			request.session?.addIpComment,
+			documentGuid,
+			request.apiClient,
+			request.currentAppeal.appealId
+		);
+
+		delete request.session.fileUploadInfo;
+
+		return response.redirect(
```

Comment:
Looks like you missed setting a notification banner on success before redirecting?

Reply:
Ah yeah i have, i'll update that now cheers!